Dockerfile

```diff
@@ -1,43 +1,68 @@
-FROM golang:1.13.5-alpine3.10
+FROM golang:1.14-buster
 
-RUN apk add --no-cache \
-    bash \
+# Remove the default $PS1 manipulation
+RUN rm /etc/bash.bashrc
+
+RUN apt-get update \
+ && apt-get -y --no-install-recommends install \
     bash-completion \
     bats \
-    coreutils \
-    curl \
-    git \
+    gettext \
     groff \
     jq \
     less \
-    openssh-client \
     python3 \
-    ncurses \
-    vim
+    python3-pip
+# ncurses-bin or libncurses6 ?
 
-WORKDIR /root/aladdin
-COPY ./commands/python/requirements.txt ./commands/python/requirements.txt
-RUN pip3 install --no-cache-dir -r ./commands/python/requirements.txt
+# Default to python3 and update setuptools
+RUN ln -fs /usr/bin/python3 /usr/bin/python \
+ && ln -fs /usr/bin/pip3 /usr/bin/pip \
+ && pip install --no-cache-dir setuptools==46.4.0
+
+# This can take a bit of time, so we do it earlier in the build process
+RUN go get -u -v sigs.k8s.io/aws-iam-authenticator/cmd/aws-iam-authenticator
 
-# update all needed tool versions here
+# Update all needed tool versions here
+ARG DOCKER_VERSION=18.09.7
 ARG KUBE_VERSION=1.15.6
-ARG KOPS_VERSION=1.15.0
 ARG HELM_VERSION=2.16.1
-ARG DOCKER_VERSION=18.09.7
+ARG KOPS_VERSION=1.15.0
+ARG POETRY_VERSION=1.0.9
+
+RUN curl -L -o- https://download.docker.com/linux/static/stable/x86_64/docker-$DOCKER_VERSION.tgz | tar -zxvf - && cp docker/docker \
+    /usr/bin/docker && chmod 755 /usr/bin/docker
 
 RUN	curl -L -o /bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v$KUBE_VERSION/bin/linux/amd64/kubectl \
 	&& chmod 755 /bin/kubectl
 
+RUN	curl -L -o- https://storage.googleapis.com/kubernetes-helm/helm-v$HELM_VERSION-linux-amd64.tar.gz | tar -zxvf - && cp linux-amd64/helm \
+	/bin/helm && chmod 755 /bin/helm && helm init --client-only
+
 RUN	curl -L -o /bin/kops https://github.com/kubernetes/kops/releases/download/$KOPS_VERSION/kops-linux-amd64 \
 	&& chmod 755 /bin/kops
 
-RUN	curl -L -o- https://storage.googleapis.com/kubernetes-helm/helm-v$HELM_VERSION-linux-amd64.tar.gz | tar -zxvf - && cp linux-amd64/helm \
-	/bin/helm && chmod 755 /bin/helm && helm init --client-only
+# On some images "sh" is aliased to "dash" which does not support "set -o pipefail".
+# We use the "exec" form of RUN to delegate this command to bash instead.
+# This is all because we have a pipe in this command and we wish to fail the build
+# if any command in the pipeline fails.
```

Comment:
Blah, this is a good point. I wonder if it wouldn't be more readable just to download to a file and run the file in that case.  It might even be less code.

Reply:
That said, even without pipefail, won't python fail if curl fails anyway?  IE, won't either part of the pipeline result in an error exit from the pipeline?

Reply:
> That said, even without pipefail, won't python fail if curl fails anyway? IE, won't either part of the pipeline result in an error exit from the pipeline?

Unfortunately no. You can see this in action with `false | python`. That results in a "success".

>  I wonder if it wouldn't be more readable just to download to a file and run the file in that case.

Probably, but at least in this manner we don't pollute the layer with the script file (not that it's that big in the first place).